### PR TITLE
Update XML for BC2 legacy items

### PR DESCRIPTION
**Pull request by BobBooth**
Sea Surface, Sea Sub Surface, and Control Measures Bc2 Legacy items added.

### CODE REVIEW

**Comment by BobBooth** [comment]
@abouffard - Added Legacy items for several Bc2 icons.

File: svg/MIL_STD_2525D_Symbols/Legacy/2525B_Change_2/S-SPNN----.svg [hunk 1]
```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (abouffard) [inline svg/MIL_STD_2525D_Symbols/Legacy/2525B_Change_2/S-SPNN----.svg]: @BobBooth  the name of this file should be S-S-NN----.svg

File: svg/MIL_STD_2525D_Symbols/Legacy/2525B_Change_2/S-SPNR----.svg [hunk 2]
```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (abouffard) [inline svg/MIL_STD_2525D_Symbols/Legacy/2525B_Change_2/S-SPNR----.svg]: @BobBooth the name for this file should be S-S-NR----.svg

File: svg/MIL_STD_2525D_Symbols/Legacy/2525B_Change_2/S-SPS-----.svg [hunk 3]
```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (abouffard) [inline svg/MIL_STD_2525D_Symbols/Legacy/2525B_Change_2/S-SPS-----.svg]: @BobBooth file name should be S-S-S-----.svg

File: svg/MIL_STD_2525D_Symbols/Legacy/2525B_Change_2/S-SPSA----.svg [hunk 4]
```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (abouffard) [inline svg/MIL_STD_2525D_Symbols/Legacy/2525B_Change_2/S-SPSA----.svg]: @BobBooth file name should be S-S-SA----.svg

File: svg/MIL_STD_2525D_Symbols/Legacy/2525B_Change_2/S-SPSP----.svg [hunk 5]
```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (abouffard) [inline svg/MIL_STD_2525D_Symbols/Legacy/2525B_Change_2/S-SPSP----.svg]: @BobBooth S-S-SP----.svg

File: svg/MIL_STD_2525D_Symbols/Legacy/2525B_Change_2/S-UPSS----.svg [hunk 6]
```diff
@@ -0,0 +1,35 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (abouffard) [inline svg/MIL_STD_2525D_Symbols/Legacy/2525B_Change_2/S-UPSS----.svg]: @BobBooth S-U-SS----.svg

File: svg/MIL_STD_2525D_Symbols/Legacy/2525B_Change_2/S-UPSSA---.svg [hunk 7]
```diff
@@ -0,0 +1,35 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (abouffard) [inline svg/MIL_STD_2525D_Symbols/Legacy/2525B_Change_2/S-UPSSA---.svg]: @BobBooth S-U-SSA---.svg

**Comment by BobBooth** [comment]
@abouffard - Thanks for your review, file names updated in latest commit.

File: instance/jmsml_D_Sea_Subsurface.xml [hunk 8]
```diff
@@ -425,6 +425,18 @@
 		    <LegacyFunctionCode Name="2525C">SO----</LegacyFunctionCode>
 	    </LegacyEntity>
     </LegacySymbol>
+    <LegacySymbol ID="MILITARY_SUBMARINE_STATION_SYM" Label="S*UPSS----*****" Remarks="Retired">
+      <LegacyFunctionCode Name="2525BC2">SS----</LegacyFunctionCode>
+      <LegacyEntity ID="MILITARY_SUBMARINE_STATION_LE" Label="WAR.SBSUF.SUB.STN" LabelAlias="Military : Submarine : Station" Graphic="S-U-SS----.svgg">
```
Comment (abouffard) [inline instance/jmsml_D_Sea_Subsurface.xml]: One too many "g" in the filename.

File: instance/jmsml_D_Sea_Subsurface.xml [hunk 9]
```diff
@@ -425,6 +425,18 @@
...
+        <LegacyFunctionCode Name="2525BC2">SS----</LegacyFunctionCode>
+      </LegacyEntity>
+    </LegacySymbol>
+    <LegacySymbol ID="MILITARY_SUBMARINE_STATION_ASW_SUBMARINE_SYM" Label="S*UPSSA---*****" Remarks="Retired">
+      <LegacyFunctionCode Name="2525BC2">SSA---</LegacyFunctionCode>
+      <LegacyEntity ID="MILITARY_SUBMARINE_STATION_ASW_SUBMARINE_SLE" Label="WAR.SBSUF.SUB.STN.ASWSUB" LabelAlias="Military : Submarine : Station : ASW : Submarine" Graphic="S-U-SSA---.svgg">
```
Comment (abouffard) [inline instance/jmsml_D_Sea_Subsurface.xml]: One too many "g" in the filename.

Comment (abouffard) [inline instance/jmsml_D_Sea_Subsurface.xml]: ID, by convention, should end in "_LE", not "_SLE".

File: instance/jmsml_D_Sea_Surface.xml [hunk 10]
```diff
@@ -1033,6 +1051,18 @@
         <LegacyFunctionCode Name="2525C">NH----</LegacyFunctionCode>
       </LegacyEntity>
     </LegacySymbol>
+    <LegacySymbol ID="MILITARY_NON_COMBAT_STATION_SYM" Label="S*SPNN---*****" Remarks="Retired">
+      <LegacyFunctionCode Name="2525BC2">NN----</LegacyFunctionCode>
+      <LegacyEntity ID="MILITARY_COMBAT_STATION_LE" Label="WAR.SSUF.NCBTT.STN" LabelAlias="Military NonCombatant : Station" Graphic="S-S-NN----.svg">
```
Comment (abouffard) [inline instance/jmsml_D_Sea_Surface.xml]: NonCombatant should be Noncombatant, to be consistent with 2525D.

File: instance/jmsml_D_Sea_Surface.xml [hunk 11]
```diff
@@ -1033,6 +1051,18 @@
...
+        <LegacyFunctionCode Name="2525BC2">NN----</LegacyFunctionCode>
+      </LegacyEntity>
+    </LegacySymbol>
+    <LegacySymbol ID="MILITARY_NON_COMBAT_STATION_RESCUE_SYM" Label="S*SPNR-----*****" Remarks="Retired">
+      <LegacyFunctionCode Name="2525BC2">NR----</LegacyFunctionCode>
+      <LegacyEntity ID="MILITARY_COMBAT_STATION_RESCUE_LE" Label="WAR.SSUF.NCBTT.STN.RSC" LabelAlias="Military NonCombatant : Station" Graphic="S-S-NR----.svg">
```
Comment (abouffard) [inline instance/jmsml_D_Sea_Surface.xml]: NonCombatant should be Noncombatant, to be consistent with 2525D.

**Comment by csmoore** [comment]
@BobBooth @abouffard - this effort/back-and-forth is to prepare to get 6 new rows to appear here - correct?  https://github.com/Esri/joint-military-symbology-xml/blob/dev/samples/legacy_support/All_ID_Mapping_Original.csv 

At what point do we see the actual row updates? Who does that?

**Comment by abouffard** [comment]
More than that.  Bob's changes will add new symbols to the ITEMS table and new entries in both lookup tables.

I'll be building a new stylx after I merge his changes.

I sent Daniel a new JMSML dll last night so he could test his changes.  I need both Bob's and Daniel's changes PR'd and merged before I build and test a final stylx for this sprint.  I'll do that this weekend.

**Comment by csmoore** [comment]
OK, so this effort/PR is to create 6 entries in one table (MappingOriginal) and 6 entries in a second table (Icons).

But sorry I'm still confused - why are @BobBooth and/or @Dbarnes1 not able to generate the legacy and style tables themselves? I thought the PEs were now able to do this themselves.

**Comment by Dbarnes1** [comment]
@csmoore I can also build and test a stylx as long as long as both @BobBooth and myself have created and merged PR's so our latest updates are in the source data.

**Comment by csmoore** [comment]
@Dbarnes1 - please ping me & I can show you how to (or git workflow for) grab the data from Bob's (or other's) fork/PR if that is what you are missing

**Comment by abouffard** [comment]
I recently set Daniel up to build CSVs and he has created stylx in the past.  I wasn't aware Bob wanted/needed to do the same, but we can quite easily set him up too, next week.

**Comment by csmoore** [comment]
It would be a good goal if regardless of who the PE is, they could start with this repo and its instructions and produce that output without someone else needing to configure their machine for them. 

Would it be possible to just include the precompiled app in this repo and have the provided bat/config files just work (unmodified) with this?  This should work for everyone - assuming there are no external dependencies.

**Comment by Dbarnes1** [comment]
@csmoore that's a great idea, I think we could add a task for next sprint to get this in to JMSML

**Comment by abouffard** [comment]
Totally concur.  We'll do that next sprint.

**Pull request merged**